stagemonitor-requestmonitor/src/main/java/org/stagemonitor/requestmonitor/MonitoredMethodRequest.java

```
@@ -49,6 +51,16 @@ public Object execute() throws Exception {
 	}
 
 	@Override
+	public ListenableFuture<Object> executeAsync() {
+		try {
+			return Futures.immediateFuture(methodExecution.execute());
+		}
+		catch (Exception e) {
+			return Futures.immediateFailedFuture(e);
```

Doesn't this swallow the exception?